ui/graphql/UpsertResolver.graphql

```diff
@@ -4,15 +4,19 @@ mutation UpsertResolver(
     $dataSourceId: Int!,
     $type: String!,
     $field: String!,
+    $preHook: String,
+    $postHook: String
     $requestMapping: String!,
-    $responseMapping: String!
+    $responseMapping: String
 ) {
     upsertResolver(
         id: $id,
         schemaId: $schemaId,
         dataSourceId: $dataSourceId,
         type: $type,
         field: $field,
+        preHook: $preHook,
```

Comment:
You need to add `preHook` and `postHook` to the result of the mutation too. Otherwise they will be emptied after save.